source/mir/blas/kernel/gemm.d

```diff
@@ -0,0 +1,91 @@
+module mir.blas.generic.gemm;
+
+import mir.internal.utility;
+
+import std.typecons: Flag;
+
+enum Conj
+{
+	none,
+	complexNone,
+	complexA,
+	complexB,
+}
+
+void gemmMicroKernel(
+	Conj conj,
+	Flag!"add" add,
+	size_t P,
+	size_t N,
+	size_t M,
+	V,
+	F
+)(
+	in size_t kc,
+	size_t blocks,
+	scope const(V[P][N])* a,
+	scope const(F[P][M])* b,
```

Comment:
does `scope` and `const` make sense together?

> This means that the destructor for an object is automatically called when the reference to it goes out of scope. The destructor is called even if the scope is exited via a thrown exception, thus scope is used to guarantee cleanup.
([scope](https://dlang.org/spec/attribute.html#scope))